scripts/genpinctrl/config.yaml

```diff
@@ -0,0 +1,136 @@
+# Configuration file for pinctrl generation (except F1 series).
+#
+# This file contains a list of pin configuration templates used to generate the
+# pinctrl files. Each entry can have the following fields:
+#
+#   - name (mandatory): This is the pin function name, e.g. UART_TX. It is used
+#     to group pin configurations alphabetically in the generated pinctrl files.
+#
+#   - match (mandatory): This is a regular expression used to match against
+#     STM32 xml database pin configuration names. The regular expression should
+#     be as precise as possible. Note that it needs to be escaped here in the
+#     configuration file.
+#
+#   - mode (mandatory): Mode setting (analog, alternate). Mode needs to
+#     be set according to the following rules:
+#       * Pin operates in analog configuration: analog
+#       * Pin operates in alternate function configuration: alternate
+#
+#   - bias (optional): Bias setting (disable, pull-up, pull-down). Equivalent to
+#     "disable" (a.k.a floating) if not set.
+#
+#   - drive (optional): Drive setting (push-pull, open-drain). Equivalent to
+#     "push-pull" if not set.
+#
+#   - slew-rate (optional): Slew rate setting (low-speed, medium-speed,
+#     high-speed, very-high-speed). Equivalent to "low-speed" if not set.
+#
+#   - variant (optional): Defines an alternative pin configuration. This is used
+#     to provide multiple configurations of a pin function (slave, master,
+#     low-power, ...).
+#
+
+---
+- name: ADC_IN / ADC_INN / ADC_INP
+  match: "^ADC(?:\\d+)?_IN[NP]?\\d+$"
+  mode: analog
+
+- name: CAN_RX
+  match: "^CAN\\d+_RX$"
+  mode: alternate
+  bias: pull-up
+
+- name: CAN_TX
+  match: "^CAN\\d+_TX$"
+  mode: alternate
+
+- name: DAC_OUT
+  match: "^DAC(?:\\d+)?_OUT\\d+$"
+  mode: analog
+
+- name: FDCAN_RX
+  match: "^FDCAN\\d+_RX$"
+  mode: alternate
+
+- name: FDCAN_TX
+  match: "^FDCAN\\d+_TX$"
+  mode: alternate
+
+- name: I2C_SCL
+  match: "^I2C\\d+_SCL$"
+  mode: alternate
+  drive: open-drain
+  bias: pull-up
+
+- name: I2C_SDA
+  match: "^I2C\\d+_SDA$"
+  mode: alternate
+  drive: open-drain
+  bias: pull-up
+
+- name: I2S_CK
+  match: "^I2S\\d+_CK$"
+  mode: alternate
+
+- name: I2S_WS
+  match: "^I2S\\d+_WS$"
+  mode: alternate
+
+- name: I2S_SD
+  match: "^I2S\\d+_SD$"
+  mode: alternate
+
+- name: SPI_MISO
+  match: "^SPI\\d+_MISO$"
+  mode: alternate
+  bias: pull-down
+
+- name: SPI_MOSI
+  match: "^SPI\\d+_MOSI$"
+  mode: alternate
+  bias: pull-down
+
+- name: SPI_SCK
+  match: "^SPI\\d+_SCK$"
+  mode: alternate
```

Comment:
To match with current code configuration:
```suggestion
- name: SPI_SCK
  match: "^SPI\\d+_SCK$"
  mode: alternate
  slew-rate: "very-high-speed"
```

Reply:
Do MISO/MOSI require `very-high-speed`?

Reply:
No, only SCK for most cases.

Reply:
As a bit per clock cycle is transmitted, slew-rate should ideally be the same on all SPI lines. For very high clock rates we could potentially see degraded data signals w.r.t SCK. CubeMX generates the following when selecting SPI:
```c
    /**SPI1 GPIO Configuration    
    PB5     ------> SPI1_MOSI
    PB4     ------> SPI1_MISO
    PB3     ------> SPI1_SCK
    PA15     ------> SPI1_NSS 
    */
    GPIO_InitStruct.Pin = GPIO_PIN_5|GPIO_PIN_4|GPIO_PIN_3;
    GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
    GPIO_InitStruct.Pull = GPIO_NOPULL;
    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
    GPIO_InitStruct.Alternate = GPIO_AF5_SPI1;
    HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);

    GPIO_InitStruct.Pin = GPIO_PIN_15;
    GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
    GPIO_InitStruct.Pull = GPIO_NOPULL;
    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
    GPIO_InitStruct.Alternate = GPIO_AF5_SPI1;
    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
```

Reply:
Current state of zephyr code is the result of what has been used, tested, reported failed and fixed on Zephyr enabled STM32 targets. So for now, I prefer we stick with same configuration used in Zephyr.

Reply:
But I'm open to change that in a latter step.

Reply:
As noted in current pinmux files (F4), this seems to be a requirement to overcome errata:

![image](https://user-images.githubusercontent.com/25011557/94279984-2bc33700-ff4d-11ea-8d39-6330527cd961.png)

I wonder if there should be a high speed `variant` for SPI_SCK for these cases. Using high slew-rates may lead to EMI problems, reflections, etc. on boards that are not properly routed (because they may not need such high speeds).

Reply:
Not sure about the speed variant. I agree this would make sense in this specific case, but I'd like to avoid having people adding their own variant for any reason.
Over variant, I'd prefer to overwrite the pin node directly in board dts in a first time. So it is done exactly in the few cases it is required.
In both cases, drawback is that people have to know he issue first before fixing it (rather than using directly the default configuration that fixes the issue). But we can have a try and see how people react (we have a full length release to get this fixed).